### PR TITLE
Add pom.xml for MLflow Spark datasource autologging integration

### PR DESCRIPTION
**Pull request by smurching**
## What changes are proposed in this pull request?

Adds a pom.xml for the Scala (JVM) side of the implementation of autologging Spark datasource information to MLflow. See this design doc for more info: https://docs.google.com/document/d/11nhwZtj-rps0stxuIioFBM9lkvIh_ua45cAFy_PqdHU/edit

We currently build our JAR against Scala 2.12 & spark-3.0-preview (as APIs for inspecting query plan info are not available in earlier Spark versions)

## How is this patch tested?

Manually running `mvn package` from `mlflow/java`,  which our tests also do, and verifying that the mlflow-spark package is built.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/java/pom.xml [hunk 1]
```diff
@@ -92,17 +93,17 @@
       <dependency>
         <groupId>com.fasterxml.jackson.dataformat</groupId>
         <artifactId>jackson-dataformat-yaml</artifactId>
-        <version>2.9.10</version>
+        <version>2.10.0</version>
```
Comment (smurching) [inline mlflow/java/pom.xml]: Bumping the Jackson databind version for compatibility with Spark 3.0 preview

File: mlflow/java/spark/pom.xml [hunk 2]
```diff
@@ -0,0 +1,141 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>mlflow-spark</artifactId>
+  <version>1.0-SNAPSHOT</version>
```
Comment (dbczumar) [inline mlflow/java/spark/pom.xml]: @smurching Should this read `1.4.1-SNAPSHOT`, or do we intend to start the new package at version 1.0?

Comment (smurching) [reply mlflow/java/spark/pom.xml]: Oops yep it should be 1.4.1-SNAPSHOT, good catch!

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @smurching

**Pull request merged**